client/events.lua

```diff
@@ -129,26 +131,22 @@ end)
 
 -- Vehicle Commands
 
----@param vehName string
-RegisterNetEvent('QBCore:Command:SpawnVehicle', function(vehName)
-    local hash = joaat(vehName)
-    if not IsModelInCdimage(hash) then return end
-    RequestModel(hash)
-    while not HasModelLoaded(hash) do
-        Wait(0)
-    end
-
+utils.entityStateHandler('initVehicle', function(entity, _, value)
+    if not value or NetworkGetEntityOwner(entity) ~= cache.playerId then return end
     if cache.vehicle then
         DeleteVehicle(cache.vehicle)
     end
 
-    local coords = GetEntityCoords(cache.ped)
-    local vehicle = CreateVehicle(hash, coords.x, coords.y, coords.z, GetEntityHeading(cache.ped), true, false)
-    TaskWarpPedIntoVehicle(cache.ped, vehicle, -1)
-    SetVehicleFuelLevel(vehicle, 100.0)
-    SetVehicleDirtLevel(vehicle, 0.0)
-    SetModelAsNoLongerNeeded(hash)
-    TriggerEvent("vehiclekeys:client:SetOwner", QBCore.Functions.GetPlate(vehicle))
+    local veh = entity
+
+    SetVehicleHasBeenOwnedByPlayer(veh, true)
+    SetNetworkIdCanMigrate(netId, true)
+    SetVehicleNeedsToBeHotwired(veh, false)
+    SetVehRadioStation(veh, 'OFF')
+    SetVehicleFuelLevel(veh, 100.0)
+    SetVehicleDirtLevel(veh, 0.0)
+    TriggerEvent("vehiclekeys:client:SetOwner", QBCore.Functions.GetPlate(veh))
+    Entity(entity).state:set('initVehicle', nil, true)
```

Comment:
```suggestion
    SetVehicleNeedsToBeHotwired(entity, false)
    SetVehRadioStation(entity, 'OFF')
    SetVehicleFuelLevel(entity, 100.0)
    SetVehicleDirtLevel(entity, 0.0)
    TriggerEvent('vehiclekeys:client:SetOwner', QBCore.Functions.GetPlate(veh))
    Entity(entity).state:set('initVehicle', nil, true)
```

Changed here as we don't need veh localized, we also can remove some natives as this is server side created vehicles.